azure-pipelines.yml

```diff
@@ -0,0 +1,36 @@
+name: Instrumented Tests
+trigger:
+- main
+
+
+pool:
+ name: DockerBuildAgents
+
+workspace:
+  clean: all
+steps:
+- script: |
+    docker --version
+    echo =============================================
+    echo Kill all running containers if existing
+    echo =============================================
+    docker container kill $(docker ps -q)
+    echo =============================================
+    echo Run unit and instrumented inside docker container
+    echo =============================================
+    docker run --privileged --cpus="3" --memory="12g" -v "$PWD":/home/gradle/ -w /home/gradle/ authclient.azurecr.io/samples/dbi-instrumented-api30 sh scripts/run-instrumented-tests.sh
+  displayName: 'Build and test inside docker container'
+- script: |
+    echo =============================================
+    echo Cleaning up build output that is owned by docker user rather than agent user
+    echo =============================================
+    docker run --privileged --cpus="3" --memory="12g" -v "$PWD":/home/gradle/ -w /home/gradle/ authclient.azurecr.io/samples/dbi-instrumented-api30 gradle clean
```

Comment:
Yeah, if the API version and/or docker target can become variables so we don't need to keep them in sync manually that would help the next person not mess it up.

Reply:
The images themselves are specific to API versions.  At some point I want to play with a matrix pipeline where the instrumented tests are run against multiple different docker images each with their own api level configuration.

Reply:
Main feedback here is "can we put the long name into a variable in the pipeline setup file so we don't have to repeat it exactly".